salt/orch/update.sls

```diff
@@ -0,0 +1,117 @@
+# Generic Updates
+update_pillar:
+  salt.function:
+    - tgt: '*'
+    - name: saltutil.refresh_pillar
+
+# TODO: When we update to 2016.11.4 uncomment this
+# update_grains:
+#   salt.function:
+#     - tgt: '*'
+#     - name: saltutil.refresh_grains
+
+update_mine:
+  salt.function:
+    - tgt: '*'
+    - name: mine.update
+    - require:
+       - salt: update_pillar
+
+update_modules:
+  salt.function:
+    - name: saltutil.sync_modules
+    - tgt: '*'
+    - kwarg:
+        refresh: True
+
+# Get list of masters needing reboot
+{%- set masters = salt.saltutil.runner('mine.get', tgt='G@roles:kube-master and G@tx_update_reboot_needed:true', fun='network.ip_addrs', tgt_type='compound') %}
+
+{%- for master_id in masters.keys() %}
+
+{{ master_id }}-clean-shutdown:
+  salt.state:
+    - tgt: {{ master_id }}
+    - sls:
+      - kubernetes-master.stop
+      - flannel.stop
```

Comment:
If that is the case, then the orchestration of masters becomes the same as the orchestration of workers, no ? I wonder then if it would make sense to simply merge both "branches" (we could handle small differences in each specific salt state or something like t that ...)